### PR TITLE
test(ARCH-482): rewrite test of datepicker using RTL

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

In the context of React 18 we need to get ride of enzyme

**What is the chosen solution to this problem?**

Rewrite test of date picker using RTL. 
Now we have 248 tests suites (from 287)

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/DateTimePickers/DateTime/Manager/Manager.component.test.js [hunk 1]
```diff
@@ -1,236 +1,245 @@
...
-							<DateTimeConsumer />
-						</Manager>,
-					);
-
-					// when
-					act(() => {
-						wrapper
-							.find('DateTimeConsumerDiv')
-							.props()
-							.onDateChange(event, { date: expectedDate, textInput, errors: [] });
```
Comment (jmfrancois) [inline packages/components/src/DateTimePickers/DateTime/Manager/Manager.component.test.js:196]: oh my god. calling onchange by hand with expectedDate and check the arg is expected :)

**Review by aamalric-talend — APPROVED** [review]
LGTM

File: packages/components/src/DateTimePickers/DateRange/Picker/Picker.component.test.js [hunk 2]
```diff
@@ -20,25 +20,19 @@ describe('DateRange.Picker', () => {
...
-			manageFocus: true,
-			onSubmit: managerValue.pickerManagement.onStartChange,
-			other: 'custom props',
-			selectedDate: new Date(2007, 0, 2),
-			endDate: new Date(2007, 1, 2),
-			t: expect.any(Function),
-			focusedInput: 'startDate',
-		});
+		expect(screen.getByLabelText('Date picker')).toBeVisible();
+		expect(container.firstChild).toMatchSnapshot();
```
Comment (ybaskaran) [inline packages/components/src/DateTimePickers/DateRange/Picker/Picker.component.test.js:32]: do we need snapshot testing ?
they have significant disadvantages, TMC is completely getting rid of snapshot testing

Comment (jmfrancois) [reply packages/components/src/DateTimePickers/DateRange/Picker/Picker.component.test.js:32]: snapshot is only good at one thing: keeping a state. I think this is still good to have one.
look at the test purpose: should render.
I don;t think you can have other purpose with snapshot. You can't write 
```javascript
it('should do this and that in the context of foo', () => {
  snapshot(render(<Component foo />))
}
```

File: packages/components/src/DateTimePickers/InputDateTimeRangePicker/InputDateTimeRangePicker.component.test.js [hunk 3]
```diff
@@ -37,36 +39,38 @@ describe('InputDateTimeRangePicker', () => {
 				}}
 			/>,
 		);
+		const start = screen.getByTestId('range-start');
+		const end = screen.getByTestId('range-end');
 
 		// then
-		expect(toJson(wrapper)).toMatchSnapshot();
```
Comment (jmfrancois) [inline packages/components/src/DateTimePickers/InputDateTimeRangePicker/InputDateTimeRangePicker.component.test.js:42]: this one for example doesn t make sens you don't look at the value

**Review by ybaskaran — APPROVED** [review]

**Pull request merged**